couchbase/src/main/java/org/hibernate/ogm/datastore/couchbase/dialect/backend/json/impl/CouchBaseDocument.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchbase.dialect.backend.json.impl;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.ogm.datastore.couchbase.logging.impl.Log;
+import org.hibernate.ogm.datastore.couchbase.logging.impl.LoggerFactory;
+import org.hibernate.ogm.datastore.couchbase.util.impl.Identifier;
+import org.hibernate.ogm.datastore.document.backend.json.impl.AbstractDocument;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.spi.Tuple;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+
+/**
+ * @author Stawicka Ewa
+ */
+public class CouchBaseDocument extends AbstractDocument {
```

Comment:
CouchBase doesn't have support for versioning documents, so I think it doean't make sense to keep revision field in the instances of CouchBase documents.